### DISCUSSION TITLE
A design question about FD usages

### DISCUSSION THREAD

**Discussion by zefengsysu** [Tencent/MMKV#1154] [General]
MMKV use two FD for each MMKV instance now, one for the MMKV file, and another for the meta file, and MMKV's file lock implementation reuse the meta file's FD. 
Since close FD after mmap return is ok, can we close the MMKV file's FD after new MemoryFile instance ready, for reduce the overhead of FD?

**Comment by lingol** [comment]
There's a report that on iOS closing FD will make mmap fail to work inter-process. Yet I don't have the time to verify that.

**Reply by zefengsysu** [reply]
MMKV does not need  inter-process mmap support on iOS, is it right? So maybe i can close the FD ASAP？

**Reply by lingol** [reply]
No, we need it. Besides, if that situation is true on iOS,  it's possible that also the same for other platforms.

**Reply by lingol** [reply]
Nevertheless, closing FD on single process mode MMKV is doable.

**Reply by zefengsysu** [reply]
Why we need the inter-process mmap support on iOS, for App Extensions?

**Reply by lingol** [reply]
Not just for App Extensions. Apps with the same [App Group](https://developer.apple.com/documentation/xcode/configuring-app-groups) could also use MMKV to share data.

**Reply by zefengsysu** [reply]
Got it. Thanks for you reply.

**Comment by lingol** [comment]
The FD usage has largely reduced in v2.2.0.
